workspaces/ballerina/ballerina-extension/src/rpc-managers/bi-diagram/rpc-manager.ts

```diff
@@ -1817,6 +1820,52 @@ export class BiDiagramRpcManager implements BIDiagramAPI {
                 });
         });
     }
+
+    async deleteType(params: DeleteTypeRequest): Promise<DeleteTypeResponse> {
+        return new Promise((resolve, reject) => {
+            const projectUri = StateMachine.context().projectUri;
+            const filePath = path.join(projectUri, params.component.filePath);
+            StateMachine.langClient().deleteType({ filePath: filePath, component: params.component})
+                .then(async (deleteTypeResponse: DeleteTypeResponse) => {
+                    console.log(">>> delete type response", deleteTypeResponse);
+                    if (deleteTypeResponse.textEdits) {
+                        await updateSourceCode({ textEdits: deleteTypeResponse.textEdits });
+                        resolve(deleteTypeResponse);
+                    } else {
+                        console.log(">>> error deleting type", deleteTypeResponse?.errorMsg);
+                        resolve(undefined);
```

Comment:
instead of resolving as undefined, we can reject with the error msg, so we can track this from the visualizer and handle on error scenario